internal/cache/gache/option_test.go

```diff
@@ -19,462 +19,372 @@ package gache
 	}
+)
+
+func TestDefaultOptions(t *testing.T) {
+	type args struct{}
 	type want struct {
-		obj *T
-		// Uncomment this line if the option returns an error, otherwise delete it
-		// err error
+		want []Option
 	}
 	type test struct {
-		name string
-		args args
-		want want
-		// Use the first line if the option returns an error. otherwise use the second line
-		// checkFunc  func(want, *T, error) error
-		// checkFunc  func(want, *T) error
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, []Option) error
 		beforeFunc func(args)
 		afterFunc  func(args)
 	}
 
-	// Uncomment this block if the option returns an error, otherwise delete it
-	/*
-	   defaultCheckFunc := func(w want, obj *T, err error) error {
-	       if !errors.Is(err, w.err) {
-	           return errors.Errorf("got error = %v, want %v", err, w.err)
-	       }
-	       if !reflect.DeepEqual(obj, w.obj) {
-	           return errors.Errorf("got = %v, want %v", obj, w.obj)
-	       }
-	       return nil
-	   }
-	*/
-
-	// Uncomment this block if the option do not returns an error, otherwise delete it
-	/*
-	   defaultCheckFunc := func(w want, obj *T) error {
-	       if !reflect.DeepEqual(obj, w.obj) {
-	           return errors.Errorf("got = %v, want %v", obj, w.c)
-	       }
-	       return nil
-	   }
-	*/
+	defaultCheckFunc := func(w want, got []Option) error {
+		if !reflect.DeepEqual(got, w.want) {
+			return errors.Errorf("got = %v, want = %v", got, w.want)
+		}
+		return nil
+	}
 
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       args: args {
-		           g: nil,
-		       },
-		       want: want {
-		           obj: new(T),
-		       },
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           args: args {
-		           g: nil,
-		           },
-		           want: want {
-		               obj: new(T),
-		           },
-		       }
-		   }(),
-		*/
+		func() test {
+			opt := []Option{
+				WithGache(gache.New()),
+			}
+			return test{
+				name: "set sucess",
+				want: want{
+					want: opt,
+				},
+				checkFunc: func(w want, got []Option) error {
+					if len(w.want) != len(got) {
+						return errors.Errorf("got length = %d want length = %d", len(got), len(w.want))
+					}
+					for i := range w.want {
+						if reflect.ValueOf(w.want[i]).Pointer() != reflect.ValueOf(got[i]).Pointer() {
+							return errors.Errorf("got = %v, want = %v", got, w.want)
+						}
+					}
+					return nil
+				},
+			}
+		}(),
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(tt *testing.T) {
-			defer goleak.VerifyNone(t)
+			defer goleak.VerifyNone(tt, goleakIgnoreOptions...)
 			if test.beforeFunc != nil {
 				test.beforeFunc(test.args)
 			}
 			if test.afterFunc != nil {
 				defer test.afterFunc(test.args)
 			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+			got := defaultOptions()
```

Comment:
```suggestion
			type args struct{}
			type want struct {
				want *cache
			}
			type test struct {
				name       string
				args       args
				want       want
				checkFunc  func(want, *cache) error
				beforeFunc func(args)
				afterFunc  func(args)
			}

			defaultCheckFunc := func(w want, cache) error {
				// please check field with go-cmp
				// e.g.) https://github.com/vdaas/vald/blob/feature/internal/tensorflow_warmup/internal/core/converter/tensorflow/tensorflow_test.go#L51-L70
				return nil
			}
			g := new(cache)
			for _, opt := range defaultOpts() {
				if err := opt(g); err != nil {
					tt.Fatal(err)
				}
			}
			if err := test.checkFunc(test.want, g); err != nil {
				tt.Errorf("error = %v", err)
			}
```

Reply:
@hlts2 
what is `defaultOpts()` ? 🤔 
I cannot find the definition about it 😓